### PR TITLE
Solution to the Reorder List Problem in java(143)

### PR DESCRIPTION
**Pull request by Monachaudhary**
Java solution to the Reorder List problem 
Time Complexity- 0(N)
Kindly review the code and merge this PR.

### CODE REVIEW

**Comment by md-shamim-ahmad** [comment]
@Monachaudhary your code does not work

**Comment by Monachaudhary** [comment]
@md-shamim-ahmad I made some changes. Can you review it.

**Comment by md-shamim-ahmad** [comment]
@Monachaudhary Hey check your code correctness from [this link](https://leetcode.com/problems/4sum/)

**Comment by Monachaudhary** [comment]
@md-shamim-ahmad I have examined it .
image

**Comment by Monachaudhary** [comment]
@md-shamim-ahmad You provided me another problem link (4 sum)

File: LeetCode Problems/101 to 200/143. Reorder List/ReorderList.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
+public class ReorderList {
```
Comment (md-shamim-ahmad) [inline LeetCode Problems/101 to 200/143. Reorder List/ReorderList.java]: class name will be 
```cpp
public class Solution or class Solution
```
not 
```cpp
public class ReorderList
```

**Review by md-shamim-ahmad — CHANGES_REQUESTED** [review]
format of this code does not follow ```clang-fromat```
so format your code in ```clang-format``` style

**Review by md-shamim-ahmad — CHANGES_REQUESTED** [review]
change the file name ```ReorderList.java``` to ```143. Reorder List.java```

**Review by md-shamim-ahmad — CHANGES_REQUESTED** [review]
added a ```README.md``` file by following this [guidelines](https://github.com/md-shamim-ahmad/LeetCode-Problem-Solution/blob/main/Contribution%20Guidelines/How%20to%20Add%20LeetCode%20Question.md)

**Comment by Monachaudhary** [comment]
I appreciate your suggestion. I will make these changes

**Comment by Monachaudhary** [comment]
I have created README.md file and made all the desired changes.

File: LeetCode Problems/101 to 200/143. Reorder List/Reorder List.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        ListNode nxt = null;
+        while(current!=null){
+            nxt = current.next;
+            current.next = prev;
+            prev= current;
+            current= nxt;
+        }
+        return prev;
+    }
+}
```
Comment (md-shamim-ahmad) [inline LeetCode Problems/101 to 200/143. Reorder List/Reorder List.java]: use this code instead your code
```java
class Solution {
    public void reorderList(ListNode head) {
        ListNode fast = head;
        ListNode slow = head;
        while (fast != null && fast.next != null) {
            fast = fast.next.next;
            slow = slow.next;
        }
        //divide the linked list into 2 halfs
        //reversing second half
        ListNode second = reverse(slow.next);
        slow.next = null;
        ListNode first = head;
        //combining both the lists
        while (second != null) {
            ListNode temp1 = first.next;
            ListNode temp2 = second.next;
            first.next = second;
            second.next = temp1;
            first = temp1;
            second = temp2;
        }
    }
    private ListNode reverse(ListNode head) {
        ListNode prev = null;
        ListNode current = head;
        ListNode nxt = null;
        while (current != null) {
            nxt = current.next;
            current.next = prev;
            prev = current;
            current = nxt;
        }
        return prev;
    }
}
```

**Review by md-shamim-ahmad — CHANGES_REQUESTED** [review]
change file name ```Reorder List.java``` to ```143. Reorder List.java``` 
@Monachaudhary

**Comment by Monachaudhary** [comment]
Thanks for your suggestion. I made changes according to you.

**Comment by md-shamim-ahmad** [comment]
@Monachaudhary hey
[solve this issue](https://github.com/md-shamim-ahmad/LeetCode-Problem-Solution/pull/81#pullrequestreview-1657262599)

**Comment by Monachaudhary** [comment]
I modified as per your suggestion

**Review by md-shamim-ahmad — APPROVED** [review]

**Comment by md-shamim-ahmad** [comment]
Thanks for your contribution :)

**Pull request merged**